src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -221,12 +218,19 @@ def save_registration_data(
                     submission, options, documents_client
                 )
 
-            if not registration_data.attachment_urls:
-                # TODO turn attachments into a dictionary when giving users more options than
-                # just urls.
-                registration_data.attachment_urls = register_submission_attachments(
-                    submission, options, documents_client
-                )
+            if options["informatieobjecttype_attachment"]:
+                for attachment in submission.attachments:
+                    try:
+                        ObjectsAPISubmissionAttachment.objects.get(
+                            submission_file_attachment=attachment
+                        )
+                    except ObjectsAPISubmissionAttachment.DoesNotExist:
+                        ObjectsAPISubmissionAttachment.objects.create(
+                            submission_file_attachment=attachment,
+                            documents_url=register_submission_attachment(
+                                submission, attachment, options, documents_client
+                            ),
+                        )
```

Comment:
If I understand correctly, this prevents that if the Objects API registration fails, the attachments are sent again to the Documents API during the retry flow?